src/App.js

```diff
@@ -1,11 +1,31 @@
+import { useState, useEffect } from 'react'
+import { loadFromLocal, saveToLocal } from './utils/localStorage'
+import locationsData from './locations.json'
 import { Route, Switch } from 'react-router-dom'
 import styled from 'styled-components/macro'
 import Navigation from './components/Navigation/Navigation'
 import HomePage from './pages/HomePage/HomePage'
 import RestaurantPage from './pages/RestaurantPage/RestaurantPage'
 import Sightseeingpage from './pages/Sightseeingpage/SightseeingPage'
+import FavoritePage from './pages/FavoritePage/FavoritePage'
+import produce from 'immer'
 
 export default function App() {
+  const [locations, setLocations] = useState(
+    loadFromLocal('locations') ?? locationsData
+  )
+  const restaurants = locations.filter(
+    location => location.type === 'restaurant'
+  )
+  const sightseeing = locations.filter(
+    location => location.type === 'sightseeing'
+  )
+  const favorites = locations.filter(location => location.isFavorite)
```

Comment:
```suggestion
  const favorites = locations.filter(({isFavorite}) => isFavorite)
```

eventually you could go with this aswell: 
```suggestion
  const favorites = locations.filter(({isFavorite}))
```